docs/source/general/new_task.rst

```diff
@@ -0,0 +1,177 @@
+******************************
+Adding a new task from scratch
+******************************
+
+One should start by subclassing :class:`~flash.core.model.Task`.
+
+To make this new task work for training, one should have to override the forward function.
+
+.. code-block:: python
+
+    from flash.core.model import Task
+
+    class ImageClassifier(Task):
+
+        def __init__(
+            self,
+            num_classes,
+            backbone="resnet18",
+            loss_fn: Callable = F.cross_entropy,
+        ):
+            super().__init__(
+                model=None,
+                loss_fn=loss_fn,
+            )
+
+            self.save_hyperparameters()
+
+            self.backbone, num_features = instantiate_backbone(backbone)
+
+            self.head = nn.Sequential(
+                nn.AdaptiveAvgPool2d((1, 1)),
+                nn.Flatten(),
+                nn.Linear(num_features, num_classes),
+            )
+
+        def forward(self, x) -> Any:
+            x = self.backbone(x)
+            return self.head(x)
+
+    model = ImageClassifier(2)
+    model(torch.randn(1, 3, 32, 32))
+
+To make this new task work for inference, one would have to create a DataPipeline.
+A shell `flash.core.data.datapipeline.DataPipeline` exposes 6 hooks to override.
+
+.. code-block:: python
+
+    class DataPipeline:
+        This class purpose is to facilitate the conversion of raw data to processed or batched data and back.
+        Several hooks are provided for maximum flexibility.
+
+        collate_fn:
+            - before_collate
+            - collate
+            - after_collate
+
+        uncollate_fn:
+            - before_uncollate
+            - uncollate
+            - after_uncollate
```

Comment:
```suggestion
        """
        This class purpose is to facilitate the conversion of raw data to processed or batched data and back.
        Several hooks are provided for maximum flexibility.

        collate_fn:
            - before_collate
            - collate
            - after_collate

        uncollate_fn:
            - before_uncollate
            - uncollate
            - after_uncollate
        """
```
this is the reason why we shall use rather `.. testcode::` so we would avoid any syntax issues